### ISSUE TITLE
Tests fail on 32-bit i586 Linux

### ISSUE THREAD

**Issue by shlomif** [petdance/html-tidy5#10]
Hi! `cpan -i HTML::Tidy5` with libtidy upgraded to 5.6.0 on mageia v6 i586, gives me these failures:

```
Loading internal null logger. Install Log::Log4perl for logging messages
CPAN: Storable loaded ok (v2.53_02)
Reading '/root/.local/share/.cpan/Metadata'
  Database was generated on Wed, 06 Jun 2018 16:17:02 GMT
Running install for module 'HTML::Tidy5'
CPAN: Digest::SHA loaded ok (v5.95_01)
CPAN: Compress::Zlib loaded ok (v2.068001)
Checksum for /root/.local/share/.cpan/sources/authors/id/P/PE/PETDANCE/HTML-Tidy5-1.04.tar.gz ok
CPAN: File::Temp loaded ok (v0.2304)
CPAN: YAML loaded ok (v1.15)
CPAN: CPAN::Meta::Requirements loaded ok (v2.132)
CPAN: Parse::CPAN::Meta loaded ok (v1.4414_001)
CPAN: CPAN::Meta loaded ok (v2.150001)
CPAN: Module::CoreList loaded ok (v5.20170114_22)
Configuring P/PE/PETDANCE/HTML-Tidy5-1.04.tar.gz with Makefile.PL
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for HTML::Tidy5
Writing MYMETA.yml and MYMETA.json
  PETDANCE/HTML-Tidy5-1.04.tar.gz
  /usr/bin/perl5.22.3 Makefile.PL -- OK
Running make for P/PE/PETDANCE/HTML-Tidy5-1.04.tar.gz
cp lib/HTML/Tidy5/Message.pm blib/lib/HTML/Tidy5/Message.pm
cp lib/HTML/Tidy5.pm blib/lib/HTML/Tidy5.pm
cp lib/Test/HTML/Tidy5.pm blib/lib/Test/HTML/Tidy5.pm
Running Mkbootstrap for HTML::Tidy5 ()
chmod 644 "Tidy5.bs"
"/usr/bin/perl5.22.3" "/usr/lib/perl5/5.22.3/ExtUtils/xsubpp"  -typemap "/usr/lib/perl5/5.22.3/ExtUtils/typemap"  Tidy5.xs > Tidy5.xsc && mv Tidy5.xsc Tidy5.c
gcc -c  -I. -I/usr/include/tidy -I/usr/local/include/tidy -I/usr/include/tidy -D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fomit-frame-pointer -march=i586 -mtune=generic -fasynchronous-unwind-tables -Wl,--as-needed -Wl,-z,relro -Wl,-O1 -Wl,--build-id -Wl,--enable-new-dtags -pthread   -DVERSION=\"1.04\" -DXS_VERSION=\"1.04\" -fPIC "-I/usr/lib/perl5/5.22.3/i386-linux-thread-multi/CORE"   Tidy5.c
rm -f blib/arch/auto/HTML/Tidy5/Tidy5.so
gcc  -shared -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fomit-frame-pointer -march=i586 -mtune=generic -fasynchronous-unwind-tables -Wl,--as-needed -Wl,-z,relro -Wl,-O1 -Wl,--build-id -Wl,--enable-new-dtags -pthread -L/usr/local/lib -fstack-protector-strong Tidy5.o  -o blib/arch/auto/HTML/Tidy5/Tidy5.so 	\
   -ltidy -lperl  	\

chmod 755 blib/arch/auto/HTML/Tidy5/Tidy5.so
"/usr/bin/perl5.22.3" -MExtUtils::Command::MM -e 'cp_nonempty' -- Tidy5.bs blib/arch/auto/HTML/Tidy5/Tidy5.bs 644
cp bin/webtidy5 blib/script/webtidy5
"/usr/bin/perl5.22.3" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/webtidy5
Manifying 3 pod documents
  PETDANCE/HTML-Tidy5-1.04.tar.gz
  /usr/bin/make -- OK
Running make test
Running Mkbootstrap for HTML::Tidy5 ()
chmod 644 "Tidy5.bs"
PERL_DL_NONLAZY=1 PERL_USE_UNSAFE_INC=1 "/usr/bin/perl5.22.3" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
# Testing HTML::Tidy5 1.04, tidy 5.6.0, Perl 5.022003, /usr/bin/perl5.22.3
t/00-load.t ................ ok
t/cfg-for-parse.t .......... ok
t/clean-crash.t ............ ok

#   Failed test '$tidy->clean("") returns empty HTML document'
#   at t/clean.t line 41.
#          got: '<!DOCTYPE html>
# <html>
# <head>
# <meta name="generator" content="HTML Tidy for HTML5 for Linux/x86 version 5.6.0">
# <title></title>
# </head>
# <body>
# </body>
# </html>
# '
#     expected: '<!DOCTYPE html>
# <html>
# <head>
# <meta name="generator" content="TIDY">
# <title></title>
# </head>
# <body>
# </body>
# </html>
# '
# Looks like you failed 1 test of 3.
t/clean.t ..................
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/3 subtests
t/extra-quote.t ............ ok
t/html_fragment_tidy_ok.t .. ok
t/html_tidy_ok.t ........... ok
t/ignore-text.t ............ ok
t/ignore.t ................. ok
t/illegal-options.t ........ ok
t/levels.t ................. ok
t/message.t ................ ok
t/opt-00.t ................. ok
t/parse-errors.t ........... ok
t/parse.t .................. ok
t/perfect.t ................ ok

#   Failed test 'Cleaned up properly'
#   at t/roundtrip.t line 36.
#          got: '<!DOCTYPE html>
# <html>
# <head>
# <meta name="generator" content="HTML Tidy for HTML5 for Linux/x86 version 5.6.0">
# <title></title>
# </head>
# <body>
# <a href="http://www.example.com/"><em>This is a test.</em></a>
# </body>
# </html>
# '
#     expected: '<!DOCTYPE html>
# <html>
# <head>
# <meta name="generator" content="TIDY">
# <title></title>
# </head>
# <body>
# <a href="http://www.example.com/"><em>This is a test.</em></a>
# </body>
# </html>
# '
# Looks like you failed 1 test of 3.
t/roundtrip.t ..............
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/3 subtests
t/segfault-form.t .......... ok
t/simple.t ................. ok
t/too-many-titles.t ........ ok
t/unicode-nbsp.t ........... ok

    #   Failed test 'Cleanup didn't break anything'
    #   at t/unicode.t line 41.
Wide character in print at /usr/lib/perl5/5.22.3/Test/Builder.pm line 1826, <DATA> line 10.
    #          got: '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 3.2//EN">
    # <html>
    # <head>
    # <meta name="generator" content="HTML Tidy for HTML5 for Linux/x86 version 5.6.0">
    # <title>日本語のホムページ</title>
    # </head>
    # <body>
    # <p>Unicodeが好きですか?</p>
    # </body>
    # </html>
    # '
    #     expected: '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 3.2//EN">
    # <html>
    # <head>
    # <meta name="generator" content="TIDY">
    # <title>日本語のホムページ</title>
    # </head>
    # <body>
    # <p>Unicodeが好きですか?</p>
    # </body>
    # </html>
    # '
    # Looks like you failed 1 test of 8.

#   Failed test 'utf8 testing'
#   at t/unicode.t line 52.

    #   Failed test 'Cleanup didn't break anything'
    #   at t/unicode.t line 66.
Wide character in print at /usr/lib/perl5/5.22.3/Test/Builder.pm line 1826, <DATA> line 10.
    #          got: '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 3.2//EN">
    # <html>
    # <head>
    # <meta name="generator" content="HTML Tidy for HTML5 for Linux/x86 version 5.6.0">
    # <title>日本語のホムページ</title>
    # </head>
    # <body>
    # <p>Unicodeが好きですか?</p>
    # </body>
    # </html>
    # '
    #     expected: '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 3.2//EN">
    # <html>
    # <head>
    # <meta name="generator" content="TIDY">
    # <title>日本語のホムページ</title>
    # </head>
    # <body>
    # <p>Unicodeが好きですか?</p>
    # </body>
    # </html>
    # '
    # Looks like you failed 1 test of 3.

#   Failed test 'Try send bytes to clean method.'
#   at t/unicode.t line 67.
# Looks like you failed 2 tests of 2.
t/unicode.t ................
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/2 subtests
t/venus.t .................. ok
t/version.t ................ ok
t/wordwrap.t ............... ok

Test Summary Report
-------------------
t/clean.t                (Wstat: 256 Tests: 3 Failed: 1)
  Failed test:  3
  Non-zero exit status: 1
t/roundtrip.t            (Wstat: 256 Tests: 3 Failed: 1)
  Failed test:  3
  Non-zero exit status: 1
t/unicode.t              (Wstat: 512 Tests: 2 Failed: 2)
  Failed tests:  1-2
  Non-zero exit status: 2
Files=25, Tests=85,  3 wallclock secs ( 0.11 usr  0.05 sys +  1.33 cusr  0.30 csys =  1.79 CPU)
Result: FAIL
Failed 3/25 test programs. 4/85 subtests failed.
Makefile:1074: recipe for target 'test_dynamic' failed
make: *** [test_dynamic] Error 255
  PETDANCE/HTML-Tidy5-1.04.tar.gz
  /usr/bin/make test -- NOT OK
//hint// to see the cpan-testers results for installing this module, try:
  reports PETDANCE/HTML-Tidy5-1.04.tar.gz

```

**Comment by petdance** [comment]
Fixed in 23db620.

**Comment by shlomif** [comment]
On Wed, 06 Jun 2018 18:40:56 -0700
Andy Lester <notifications@github.com> wrote:

> Fixed in 23db620.
> 

Thanks!

-- 
-----------------------------------------------------------------
Shlomi Fish       http://www.shlomifish.org/
Optimising Code for Speed - http://shlom.in/optimise

Windows Update does not take forever to finish updating your computer. It takes
at least forever and two weeks.
    — http://www.shlomifish.org/humour/bits/facts/Windows-Update/

Please reply to list if it's a mailing list post - http://shlom.in/reply .
